ruby-service/app/services/cache_manager.rb

```
@@ -0,0 +1,101 @@
+      history_size: @cache_history.length
+    }
+  end
+
+  def get_keys_by_pattern(pattern)
+    @cache.keys.select { |k| k.to_s.match?(pattern) }
+  end
+
+  private
+
+  def evict_lru
+    oldest_key = @access_times.min_by { |k, v| v }&.first
```

RuboCop (1.88.2)

[warning] Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.

(Lint/UnusedBlockArgument)